### PR TITLE
ramips: add missing reset button for Nexx WT1520

### PR DESCRIPTION
**Pull request by musashino205**
This commit adds missing the GPIO key used as reset button.
Nexx WT1520 has a GPIO key for factory reset, but it's not defined in
WT1520.dtsi and cannot use it.

GPIO10: reset button

Signed-off-by: INAGAKI Hiroshi <musashino.open@gmail.com>

Compile tested: WT1520 (4M)
Run tested: WT1520 (4M)

### CODE REVIEW

**Comment by mkresin** [comment]
Do you have a serial console attached?

`bootargs = "console=ttyS1,57600";` indicates that that the 2nd uart (uart full) is used as serial console. If you switch the uartf_pins to gpio, the serial console will not work any longer (which I consider more important than the reset button).

Having a look at the OEM bootlog linked in the OpenWrt wiki, the 2nd uart us used there as well:

> Kernel command line: console=ttyS1,57600n8 root=/dev/ram0
> serial8250: ttyS1 at I/O 0xb0000c00 (irq = 12) is a 16550A

It is possible to switch the uartf pins into a "gpio uartf" mode, where the 2nd UART will work like an uart lite:

| Pin Name/Pinmux mode | "uartf" | "gpio uartf" |  "gpio" |
| ------ | ----------- |  ----------- |  ----------- |
| RIN | RIN | GPIO#14 | GPIO#14 | 
| DSR_N | DSR_N | GPIO#13 | GPIO#13 | 
| DCD_N | DCD_N | GPIO#12 | GPIO#12 |
| DTR_N | DTR_N | GPIO#11 | GPIO#11 |
| RXD  | RXD | RXD | GPIO#10 |
| CTS_N | CTS_N |CTS_N | GPIO#9 |
| TXD | TXD | TXD | GPIO#8 |
| RTS_N | RTS_N | RTS_N | GPIO#7 |

But even in that mode it isn't possible to use GPIO#10.

It rather looks like the reset button doesn't work with the OEM firmware either or the reset button is on another GPIO. Might be that even more weird stuff is going on.

**Comment by musashino205** [comment]
Yes, I have a serial console attached.

Hmm... As of now, both serial console and the reset button are working properly in my WT1520 (4M).
Also, the reset button was also working with OEM firmware.

- serial console screenshot
lede_wt1520_console

- reset button pressed
lede_wt1520_reset-console

**Comment by mkresin** [comment]
Would it be possible that you provide a full bootlog with your changes applied. Might be that the chosen bootargs from the WT1520.dtsi are ignored at all and the expected uartlite (ttyS0) is used for the serial output.

Would you please do another test with the chosen node dropped from the WT1520.dtsi.

**Comment by musashino205** [comment]
- bootlog
https://gist.github.com/musashino205/a02d637be4c4315212a345467eb60cbc

> Would you please do another test with the chosen node dropped from the WT1520.dtsi.

OK, I'll try it later.

**Comment by musashino205** [comment]
I tried the test with the chosen node dropped, but serial console stopped duaring boot.
- bootlog
https://gist.github.com/musashino205/a8395b3c11bb16919d0dd86e957bb650

**Comment by mkresin** [comment]
> I tried the test with the chosen node dropped, but serial console stopped duaring boot.

Okay, I see what is going on. The UARTs are registered in a different order to what I expected to the system:

>[    0.521310] 10000500.uart: ttyS0 at MMIO 0x10000500 (irq = 13, base_baud = 2500000) is a Palmchip BK-3103
>[ 0.541523] 10000c00.uartlite: ttyS1 at MMIO 0x10000c00 (irq = 20, base_baud = 2500000) is a Palmchip BK-3103

The - never used - uartfull is registered as ttyS0 in the system, the `bootargs = "console=ttyS1,57600";` kernel commandline is required, to use the uartlite as serial console.

Means, it is perfect fine to switch the UARTF pins to GPIO. But the never used uartfull should be dropped from the WT1520.dtsi.

Long story short, drop the `&uart { };` node and the `chosen` node from the WT1520.dtsi

Afterwards the uartlite should be at ttyS0 and you can use (in fact you will be using) the kernel commandline from rt5350.dtsi.

Please confirm/deny that my assumptions are correct. If they are correct, please update the PR.

Please use ```git commit --amend``` to add changes to your commit. Use ```git push --force``` to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

**Comment by musashino205** [comment]
I modified WT1520.dtsi as pointed out and it worked with uartlite (ttyS0) without any problems.

Thank you for the advices. It was very helpful!

**Comment by mkresin** [comment]
merged, thanks!
